aten/src/ATen/native/quantized/cpu/qlinear.cpp

```diff
@@ -900,6 +901,104 @@ at::Tensor PackedLinearWeightsOnednn:: apply_tanh(
       std::move(input), output_scale, output_zero_point);
 }
 
+static at::Tensor linear_int8_with_onednn_weight(
+    at::Tensor input, // int8 CPU Tensor, not QTensor
+    double input_scale,
+    int64_t input_zero_point,
+    at::Tensor onednn_weight, // int8 tensor from MkldnnCPU
+    at::Tensor weight_scales,
+    at::Tensor weight_zero_points,
+    c10::optional<at::Tensor> bias, // plain tensor
+    double output_scale,
+    int64_t output_zero_point,
+    bool fp32_output,
+    std::string& post_op_name, // e.g. "none", "relu"
+    torch::List<double>& post_op_args,
+    std::string& post_op_algorithm) {
+  using ideep::tensor;
+  const int64_t dim = input.dim();
+  TORCH_CHECK(input.scalar_type() == c10::ScalarType::Byte,
+      "qlinear with mkldnn tensor: data type of input should be uint8 (unsigned char).");
+  TORCH_CHECK(onednn_weight.scalar_type() == c10::ScalarType::Char,
+      "qlinear with mkldnn tensor: data type of weight should be int8 (char).");
+  TORCH_CHECK(
+      weight_scales.scalar_type() == c10::ScalarType::Float, "weight scales should be dtype c10::ScalarType::Float.");
+  if (fp32_output) {
+    TORCH_CHECK(
+        output_scale == 1.0f && output_zero_point == 0, "onednn qlinear: expect scale=1 and zero point=0 for fp32 output");
+  }
+
+  auto src = at::native::itensor_from_tensor(input);
```

Comment:
Do we need to ensure the input is in `contiguous` format before using its data ptr?

Reply:
Thanks. It's fixed.